syntax = "proto3";

option cc_generic_services = true;
package ImageService;
///////////////////////////////////////////////////////
// CImageService
///////////////////////////////////////////////////////
message ImgCntRequest
{
  string channel_name = 1;
}

message ImgCntResponse
{
  int32 total_frames = 1;
  string channel_name = 2;
}
///////////////////////////////////////////////////////
message infoChannelRequest
{
  bool required_channel_info = 1;
}

message deviceData
{
  string deviceName = 1;
  repeated channelInfo channel_Info = 2;
  int32 no_of_channels = 3;
}
///////////////////////////////////////////////////////
message channelInfo
{
  string channel_name = 1;
  repeated uint64 timestamp = 2;
}
///////////////////////////////////////////////////////
message sourceFileRequest
{
    string sourcefile_name = 1;
}
message sourceFileResp
{
    string sourcefile_name = 1;
}
///////////////////////////////////////////////////////
message setSourceFolderRequest
{
    string sourcefolder_name = 1;
}
message setSourceFolderResp
{
    string sourcefolder_name = 1;
}
///////////////////////////////////////////////////////
message devicesDataRequest
{
    bool required_devicesData = 1;
}

message devicesDataResponse
{
	repeated deviceData devicedata = 1;
	int32 deviceCount = 2;
}
///////////////////////////////////////////////////////
message PixelImageData
{
    string name = 1;
    string format = 2;
    bytes data = 3;
    uint64 width = 4;
    uint64 height = 5;
}

message ImageRequest
{
    uint64 required_timestamp = 1;
    string request_channel_name = 2;
    string request_device_name = 3;
    repeated string image_name = 4;
	uint64 image_index = 5;
}

message ImageResponse
{
    uint64 recieved_timestamp = 1;
    string response_channel_name =2;
    string base_image_name = 3;
    bytes base_image = 4;
    repeated PixelImageData image_bytes = 5;
    string response_device_name = 6;
}

message ImageDeleteRequest
{
    uint64 required_timestamp = 1;
    string request_channel_name = 2;
    repeated string image_name = 3;
}

message PixelLabelWriteRequest
{
    uint64 recieved_timestamp  = 1;
    string response_channel_name = 2;
    repeated PixelImageData image = 3;
}

message PixelLabelWriteResponse
{
    bool status = 1;
}

message HFLResponse {
	uint64 timestamp = 1;
	bytes intensity_image = 2;
	bytes distance_image = 3;
	uint64 HFL_image_index = 4;
} 